analytics_data_api/v0/views/problems.py

```diff
@@ -104,7 +112,22 @@ class ProblemResponseAnswerDistributionView(generics.ListAPIView):
     def get_queryset(self):
         """Select all the answer distribution response having to do with this usage of the problem."""
         problem_id = self.kwargs.get('problem_id')
-        return ProblemResponseAnswerDistribution.objects.filter(module_id=problem_id)
+        consolidate_variants = self.request.QUERY_PARAMS.get('consolidate_variants', 'False')
+        consolidate_variants = (consolidate_variants.lower() == 'true')
+
+        queryset = ProblemResponseAnswerDistribution.objects.filter(module_id=problem_id).order_by('part_id')
+
+        if not consolidate_variants:
+            return queryset
+
+        self.serializer_class = ConsolidatedAnswerDistributionSerializer
+        consolidated_rows = []
+
+        # pylint: disable=unused-variable
+        for part_id, part in groupby(queryset, lambda x: x.part_id):
```

Comment:
you could do `for _, part in groupby(...)` instead to avoid the pylint directive.